internal/aws/cwlogs/utils.go

```diff
@@ -68,3 +70,20 @@ func ValidateTagsInput(input map[string]*string) error {
 
 	return nil
 }
+
+// Helper function to convert a map to a consistent string representation
+func mapToString(m map[string]*string) string {
+	if m == nil {
+		return ""
+	}
+	var pairs []string
+	for k, v := range m {
+		if v == nil {
+			pairs = append(pairs, fmt.Sprintf("%s:", k))
+		} else {
```

Comment:
Another thing I am thinking about is that ordering matters and go maps are unordered.  

Chatgpt recommendation.  Not the answer but just an example.  It covers the sorting issue and the sprintf performance issue
```
// MapToString converts a map[string]*string to a consistent string representation.
// Each key-value pair is formatted as "key:value" and separated by semicolons.
// If a value is nil, it is represented as "key:".
// The pairs are sorted alphabetically by key to ensure consistency.
func MapToString(m map[string]*string) string {
    if m == nil {
        return ""
    }

    pairs := make([]string, 0, len(m))
    for k, v := range m {
        if v == nil {
            pairs = append(pairs, k+":")
        } else {
            pairs = append(pairs, k+":"+*v)
        }
    }

    sort.Strings(pairs)
    return strings.Join(pairs, ";")
}
```

Reply:
a unit test that validates stability (ordering) would also be worthwhile

Reply:
For clarity, this function was used to convert only the two maps from an Entity (`KeyAttributes` and `Attributes`) to a string for the hash function. I'm now realizing that we don't actually need this function, the Entity comes with to `GoString()` function that returns the string representation of the entity. Will remove this function altogether and edit the hash function.